packages/gatsby-cli/src/create-cli.js

```diff
@@ -0,0 +1,194 @@
+const path = require(`path`)
+const resolveCwd = require(`resolve-cwd`)
+const yargs = require(`yargs`)
+const report = require(`./reporter`)
+
+const DEFAULT_BROWSERS = [`> 1%`, `last 2 versions`, `IE >= 9`]
+
+const handlerP = fn => (...args) => {
+  Promise.resolve(fn(...args)).then(
+    () => process.exit(0),
+    err => report.panic(err)
+  )
+}
+
+function buildLocalCommands(cli, isLocalSite) {
+  const defaultHost = `localhost`
+  const directory = path.resolve(`.`)
+
+  let siteInfo = { directory, browserslist: DEFAULT_BROWSERS }
+  if (isLocalSite) {
+    const json = require(path.join(directory, `package.json`))
+    siteInfo.sitePackageJson = json
+    siteInfo.browserslist = json.browserslist || siteInfo.browserslist
+  }
+
+  function resolveLocalCommand(command) {
+    if (!isLocalSite) {
+      cli.showHelp()
+      report.verbose(`current directory: ${directory}`)
+      return report.panic(
+        `gatsby <${command}> can only be run for a gatsby site. \n` +
+          `Either the current working directory does not contain a package.json or ` +
+          `'gatsby' is not specified as a dependency`
+      )
+    }
+
+    try {
+      const cmdPath =
+        resolveCwd.silent(`gatsby/dist/commands/${command}`) ||
+        // Old location of commands
+        resolveCwd.silent(`gatsby/dist/utils/${command}`)
+      if (!cmdPath)
+        return report.panic(
+          `There was a problem loading the local ${command} command. Gatsby may not be installed.`
+        )
+
+      report.verbose(`loading local command from: ${cmdPath}`)
+      return require(cmdPath)
+    } catch (err) {
+      cli.showHelp()
+      return report.panic(
+        `There was a problem loading the local ${command} command. Gatsby may not be installed.`,
+        err
+      )
+    }
+  }
+
+  function getCommandHandler(command, handler) {
+    return argv => {
+      report.setVerbose(!!argv.verbose)
+
+      process.env.gatsby_log_level = argv.verbose ? `verbose` : `normal`
+      report.verbose(`set gatsby_log_level: "${process.env.gatsby_log_level}"`)
+
+      process.env.gatsby_executing_command === command
```

Comment:
@KyleAMathews i added these just now, similar to how npm does I think this is the easiest way to handle checks for this across package instances and the like. WDYT?

Reply:
Did you mean do an equality check here? Maybe not sure what you're doing :-)

Reply:
lol oops